engine/configurator/conf_cyaml.c

```diff
@@ -0,0 +1,258 @@
+        case CYAML_ERR_LIBYAML_PARSER:
+        /* Count of CYAML return codes. */
+        case CYAML_ERR__COUNT:
+        {
+            ERROR("There is a CYAML error %s(%d)",
+                  cyaml_strerror(err), err);
+            return TE_EINVAL;
+        }
+        default:
+        assert(0);
+    }
+}
+
+void
+free_hist_seq(history_seq hist_seq);
+
+void
+free_obj(object_type obj)
+{
+    unsigned int i;
+
+    free(obj.d);
+    free(obj.oid);
+    free(obj.def_val);
+    for (i = 0; i < obj.depends_count; i++)
+        free(obj.depends[i].oid);
+    free(obj.depends);
+}
+
+void
+free_inst(instance_type inst)
+{
+    free(inst.if_cond);
+    free(inst.oid);
+    free(inst.value);
+}
+
+
+void
+free_b_entry(backup_entry entry)
+{
+
+    if (entry.object != NULL)
+    {
+        free_obj(*entry.object);
+        free(entry.object);
+    }
+
+    if (entry.instance != NULL)
+    {
+        free_inst(*entry.instance);
+        free(entry.instance);
+    }
+}
+
+void
+free_b_seq(backup_seq b_seq)
+{
+    unsigned int i;
+
+    for (i = 0; i < b_seq.entries_count; i++)
+        free_b_entry(b_seq.entries[i]);
+    free(b_seq.entries);
+    b_seq.entries = NULL;
+    b_seq.entries_count = 0;
+}
+
+void
+free_cond_entry(cond_entry cond)
+{
+    free(cond.if_cond);
+    if (cond.then_cond != NULL)
+    {
+        free_hist_seq(*cond.then_cond);
+        free(cond.then_cond);
+    }
+    if (cond.else_cond != NULL)
+    {
+        free_hist_seq(*cond.else_cond);
+        free(cond.else_cond);
+    }
+}
+
+void
+free_hist_entry(history_entry entry)
+{
+    unsigned int i;
+
+    free(entry.comment);
+
+    if (entry.cond != NULL)
+    {
+        free_cond_entry(*entry.cond);
+        free(entry.cond);
+    }
+
+    free(entry.reboot_ta);
+
+#define FREE_SEQUENCE(cmd_, free_func_) \
+    for (i = 0; i < entry.cmd_ ## _count; i++)    \
+        free_func_(entry.cmd_[i]);                \
+    free(entry.cmd_);                             \
+    entry.cmd_ = NULL;                            \
+    entry.cmd_ ## _count = 0
+
+    FREE_SEQUENCE(incl, free);
+    FREE_SEQUENCE(reg, free_obj);
+    FREE_SEQUENCE(unreg, free_obj);
+    FREE_SEQUENCE(add, free_inst);
+    FREE_SEQUENCE(get, free_inst);
+    FREE_SEQUENCE(delete, free_inst);
+    FREE_SEQUENCE(copy, free_inst);
+    FREE_SEQUENCE(set, free_inst);
+
+#undef FREE_SEQUENCE
+}
+
+void
+free_hist_seq(history_seq hist_seq)
+{
+    unsigned int i;
+
+    for (i = 0; i < hist_seq.entries_count; i++)
+        free_hist_entry(hist_seq.entries[i]);
+    free(hist_seq.entries);
+    hist_seq.entries = NULL;
+    hist_seq.entries_count = 0;
+}
+
+/* That prints some part of debug.
+ */
+void print_backup(backup_seq *backup)
+{
+    for (unsigned int i = 0;
+         i < backup->entries_count;
+         i += (backup->entries_count-i) < 20 ? 1 : 10)
+    {
+        if (backup->entries[i].object != NULL)
+            ERROR("%u. object. %s\n", i, backup->entries[i].object->oid);
```

Comment:
Why is it `ERROR` ?

Reply:
Sorry, these temporary functions that used for debugging should be removed before my request. These functions are removed.